api/hasura/actions/_handlers/syncCoSoul.ts

```diff
@@ -106,14 +126,39 @@ const burned = async (address: string) => {
           },
         },
         {
-          __typename: true,
+          returning: {
+            token_id: true,
+          },
         },
       ],
     },
     {
       operationName: 'syncCoSoul__deleteCoSoul',
     }
   );
+
+  const burnedCosoul = delete_cosouls?.returning.pop();
+  assert(burnedCosoul);
+  const { insert_interaction_events_one } = await adminClient.mutate(
```

Comment:
Let's follow the pattern everywhere else and use the `mutations.insertInteractionEvents` helper method.